src/array/fixed_size_list.rs

```diff
@@ -343,56 +514,91 @@ mod tests {
         assert_eq!(array_nullable_string_nullable.index(3), None);
     }
 
+    #[test]
+    fn to_debug() {
+        let input = [[1_u8, 2], [3, 4]];
+        let array = input
+            .into_iter()
+            .collect::<FixedSizeListArray<2, FixedSizePrimitiveArray<u8>>>();
+        let collected = array.into_iter().collect::<Vec<_>>();
+        assert_eq!(collected, [[1, 2], [3, 4]]);
+    }
+
+    // #[test]
+    // fn into_iter_test() {
+    //     let input_nullable_string = [
+    //         Some(["hello".to_owned(), "world".to_owned()]),
+    //         None,
+    //         Some(["hello".to_owned(), "again".to_owned()]),
+    //     ];
+    //     let array_nullable_string = input_nullable_string
+    //         .clone()
+    //         .into_iter()
+    //         .collect::<FixedSizeListArray<2, StringArray<false>, true>>();
+    //     assert_eq!(
+    //         array_nullable_string.into_iter().collect::<Vec<_>>(),
+    //         input_nullable_string
+    //     );
+    // }
```

Comment:
The underlying `StringArray` seems a bit weird for the example `into_iter_test()`, in that, I would have expected only the `self.bitmap` (`vec![5]` == `[1, 0, 1]`, which is correct), corresponding to the `FixedSizeListArray`'s bitmap, to contain nullability information in this case.

But as can be seen, even the offsets of the underlying `StringArray` (`offsets：vec! [0, 5, 10, 10, 10, 15, 20]`) encode two null values after two valid ones, but, IIUC, it should just be `offsets：vec! [0, 5, 10, 15, 20]` in this case?

WDYT @mbrobbel ?

<img width="1840" alt="Screenshot 2024-05-20 at 01 49 32" src="https://github.com/mbrobbel/narrow/assets/65168627/e747d7d3-532c-4d43-9b5d-b99c90234403">

Reply:
Okay, wait, this is not limited to just strings, this test also shows the same problem -- something may be off with the trait bounds.

```rust
#[test]
    fn into_iter_test_int() {
        let input_nullable_int = [Some([1_u8, 2_u8]), None, Some([1_u8, 4_u8])];
        let array_nullable_int = input_nullable_int
            .clone()
            .into_iter()
            .collect::<FixedSizeListArray<2, FixedSizePrimitiveArray<u8, false>, true>>();
        assert_eq!(
            array_nullable_int.into_iter().collect::<Vec<_>>(),
            input_nullable_int
        );
    }
```

Reply:
Okay, I may have misunderstood the spec 🙂